auth.js

```diff
@@ -383,354 +373,360 @@ module.exports = function auth( options ) {
-            return next( err );
+        init_session(req, res, function (err) {
+          if (err) {
+            return next(err)
           }
 
-          restriction( req, res, next )
+          restriction(req, res, next)
         })
       })
     }
   }
 
-
-  function init( msg, respond ) {
+  function init (msg, respond) {
     respond()
   }
 
-
-  function authcontext( req, res, msg, act, respond ) {
+  function authcontext (req, res, msg, act, respond) {
     var user = req.seneca && req.seneca.user
-    if( user ) {
+    if (user) {
       msg.user = user
     }
 
     var login = req.seneca && req.seneca.login
-    if( login ) {
+    if (login) {
       msg.login = login
     }
 
-    act( msg, function( err, out ) {
-      if( err ) {
+    act(msg, function (err, out) {
+      if (err) {
         seneca.log.debug(err)
         out = out || {}
 
-        return respond( null, out );
+        return respond(null, out)
       }
 
-      return respond( null,out )
+      return respond(null, out)
     })
   }
 
-  var config = { prefix:options.prefix, redirects:{} }
+  var config = {prefix: options.prefix, redirects: {}}
 
-
-  function do_respond( err, action, req, next, forceStatus, forceRedirect ) {
-    req.seneca.act( "role: 'auth', cmd: 'redirect'", {kind: action}, function( errRedirect, redirect ) {
-      if( err ) {
-        if( redirect ) {
-          req.seneca.log.debug( 'redirect', 'fail', redirect.fail )
-          return next( null, {http$: {status: forceStatus || 301, redirect:forceRedirect || redirect.fail}} )
+  function do_respond (err, action, req, next, forceStatus, forceRedirect) {
+    req.seneca.act("role: 'auth', cmd: 'redirect'", {kind: action}, function (errRedirect, redirect) {
+      if (err) {
+        if (redirect) {
+          req.seneca.log.debug('redirect', 'fail', redirect.fail)
+          return next(null, {http$: {status: forceStatus || 301, redirect: forceRedirect || redirect.fail}})
         }
         else {
-          return next( null, {http$: {status: forceStatus || 200, redirect:forceRedirect}, ok: false, why: err} )
+          return next(null, {http$: {status: forceStatus || 200, redirect: forceRedirect}, ok: false, why: err})
         }
       }
-      else{
-        if( redirect ) {
-          req.seneca.log.debug( 'redirect', 'win', redirect.win )
-          return next( null, {http$: {status: forceStatus || 301,redirect:forceRedirect || redirect.win}} )
+      else {
+        if (redirect) {
+          req.seneca.log.debug('redirect', 'win', redirect.win)
+          return next(null, {http$: {status: forceStatus || 301, redirect: forceRedirect || redirect.win}})
         }
         else {
-          seneca.act( "role:'auth', cmd:'clean'", {user:req.seneca.user, login:req.seneca.login}, function( err, out ) {
+          seneca.act("role:'auth', cmd:'clean'", {user: req.seneca.user, login: req.seneca.login}, function (err, out) {
+            if (err) {
+              // handle error
```

Comment:
@mihaidma

what about this:
- log error
- return next({http$: {status: 500}})

Internal server error